src/OrchardCore/OrchardCore/Modules/ModularBackgroundService.cs

```diff
@@ -34,17 +37,25 @@ internal class ModularBackgroundService : BackgroundService
         private readonly IHttpContextAccessor _httpContextAccessor;
         private readonly ILogger _logger;
         private readonly IClock _clock;
+        private readonly BackgroundServiceOptions _options;
 
         public ModularBackgroundService(
             IShellHost shellHost,
             IHttpContextAccessor httpContextAccessor,
             ILogger<ModularBackgroundService> logger,
-            IClock clock)
+            IClock clock,
+            Microsoft.Extensions.Configuration.IConfiguration configuration)
         {
             _shellHost = shellHost;
             _httpContextAccessor = httpContextAccessor;
             _logger = logger;
             _clock = clock;
+
+            _options = configuration
```

Comment:
Shouldn’t this be registered as an IOptions?

here it’s just reading from IConfiguration directly?

The naming here looks much better thanks @jersiovic

Reply:
> Shouldn’t this be registered as an IOptions?

Yes could be and then e.g. a default config setup that would read from IConfiguration.

But as said in my last comment, this option is only used here and because it is at the app level other modules can't collaborate to set it by code, would need to be done by app level code.

So to keep it simpler I think it is okay to not use the `IConfigureOptions<>` pattern just to do the same thing (read from configuration), as we do for `DatabaseShellsStorageOptions`.

But I'm not against ;)

Reply:
Since we’ve added an Options class it would make sense to me. 

Then it is actually configurable via the class as well, otherwise the class doesn’t do anything I think?

In this way you can configure it via appsettings, or at app level in code.

not to worried either way @jtkech and @jersiovic  just saw the class and couldn’t figure why we’d add the class without registering it.

Reply:
Okay cool then, could be a simple `services.Configure<>()` in the `AddBgService()` helper.

I will suggest the few lines of code to do it (can't edit the files)

Reply:
> otherwise the class doesn’t do anything I think?

Yes and no, the class is an helper in itself when binding to a `BackgroundServiceOptions` field, so if we add more options to this class, we will not have to change the constructor code.